DatadogInternal/Tests/Swizzling/MethodSwizzlerTests.swift

```diff
@@ -41,86 +37,118 @@ class MethodSwizzlerTests: XCTestCase {
         // before
         XCTAssertNotEqual(obj.perform(selToSwizzle)?.takeUnretainedValue() as? String, String.mockAny())
         // swizzle
-        let foundMethod = try Swizzler.findMethod(with: selToSwizzle, in: BaseClass.self)
-        swizzler.swizzle(foundMethod) { currentImp -> TypedBlockIMPReturnString in
+        let foundMethod = try dd_class_getInstanceMethod(BaseClass.self, selToSwizzle)
+        swizzler.swizzle(foundMethod) { currentImp -> MethodOverride in
             return { impSelf in
                 return currentImp(impSelf, self.selToSwizzle).appending(String.mockAny())
             }
         }
         // after
         XCTAssertEqual(obj.perform(selToSwizzle)?.takeUnretainedValue() as? String, BaseClass.returnValue + String.mockAny())
+        swizzler.unswizzle()
     }
 
     func test_searchWrongSelector() {
         let wrongSelToSwizzle = Selector(("selector_who_never_existed"))
 
         let expectedErrorDescription = "\(NSStringFromSelector(wrongSelToSwizzle)) is not found in \(NSStringFromClass(BaseClass.self))"
-        XCTAssertThrowsError(try Swizzler.findMethod(with: wrongSelToSwizzle, in: BaseClass.self), "Wrong selector should throw") { error in
+        XCTAssertThrowsError(try dd_class_getInstanceMethod(BaseClass.self, wrongSelToSwizzle), "Wrong selector should throw") { error in
             let internalError = error as? InternalError
             XCTAssertEqual(internalError?.description, expectedErrorDescription)
         }
     }
 
+    func test_findSubclassMethod() throws {
+        class EmptySubclass: BaseClass { }
+        class EmptySubSubclass: EmptySubclass { }
+        XCTAssertNotNil(try dd_class_getInstanceMethod(EmptySubclass.self, selToSwizzle))
+        XCTAssertNotNil(try dd_class_getInstanceMethod(EmptySubSubclass.self, selToSwizzle))
+    }
+
     func test_swizzle_alreadySwizzledSelector() throws {
-        let foundMethod = try Swizzler.findMethod(with: selToSwizzle, in: BaseClass.self)
+        let method = try dd_class_getInstanceMethod(BaseClass.self, selToSwizzle)
+        let obj = BaseClass()
+        let before_imp = method_getImplementation(method)
```

Comment:
We need to increase the coverage here
- Make sure all implementation are called including the original implementation
 - Lets say first implementation returns "first", second returns first imp + " second", third returns second + " third", then you should have "first second third" as returned value in the end.
- unswizzling something that is in the middle, like second swizzling in the above example